### PR TITLE
Promote pod's websocket based NodeConformance tests to Conformance

### PR DESCRIPTION
**Pull request by mgdevstack**
**What this PR does / why we need it**:
Promotes pod's WebSocket based [NodeConformance] tests to [Conformance]

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #66935 

**Special notes for your reviewer**:
- For the time being, we are not promoting [SLOW] tests as they are taking a long time to run. Added a comment about slow tests.
- `should support pod readiness gates [NodeFeature:PodReadinessGate]`: PodReadinessGate feature is in Alpha so not promoting this test to Conformance.
- No Flakes found.

**Release note**:

```release-note
NONE
```
/area conformance
@kubernetes/sig-node-pr-reviews

### CODE REVIEW

**Comment by spiffxp** [comment]
FYI @kubernetes/cncf-conformance-wg 
/kind cleanup
/lgtm
/assign @bgrant0607 @smarterclayton 
for the /approve

File: test/e2e/common/pods.go [hunk 1]
```diff
@@ -29,18 +29,18 @@ import (
...
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/intstr"
 	"k8s.io/apimachinery/pkg/util/uuid"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/apimachinery/pkg/watch"
 	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
 	"k8s.io/kubernetes/pkg/kubelet"
 	"k8s.io/kubernetes/test/e2e/framework"
+	imageutils "k8s.io/kubernetes/test/utils/image"
```
Comment (smarterclayton) [inline test/e2e/common/pods.go]: Please move these changes to a separate PR (and copy me on it) so we can merge that first.  The only changes allowed in a conformance change are the ones that mark the tests conformance and the change to conformance.txt.

File: test/e2e/common/pods.go [hunk 2]
```diff
@@ -645,7 +658,7 @@ var _ = framework.KubeDescribe("Pods", func() {
 		}
 	})
 
-	// Slow issue #19027 (20 mins)
+	// Slow by design (~27 mins) issue-#19027
```
Comment (smarterclayton) [inline test/e2e/common/pods.go]: Same comment here

File: test/e2e/common/pods.go (hunk 1 above)
Comment (mgdevstack) [reply test/e2e/common/pods.go]: Thanks for pointing it. Tracking changes in https://github.com/kubernetes/kubernetes/pull/69169

File: test/e2e/common/pods.go (hunk 2 above)
Comment (mgdevstack) [reply test/e2e/common/pods.go]: Done

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve

**Comment by spiffxp** [comment]
This has been changed to just promotion, think it's ready for another look

**Comment by smarterclayton** [comment]
/lgtm

Clearly within the set of functionality expected for a cluster to implement in order to provide reasonable feedback to a user performing debugging.
